tools/github-event-processor/Azure.Sdk.Tools.GitHubEventProcessor/EventProcessing/PullRequestProcessing.cs

```diff
@@ -0,0 +1,245 @@
+using Octokit;
+using System.Threading.Tasks;
+using Azure.Sdk.Tools.GitHubEventProcessor.GitHubPayload;
+using Azure.Sdk.Tools.GitHubEventProcessor.Utils;
+using Azure.Sdk.Tools.GitHubEventProcessor.Constants;
+using System.Text.Json;
+using Octokit.Internal;
+
+namespace Azure.Sdk.Tools.GitHubEventProcessor.EventProcessing
+{
+    public class PullRequestProcessing
+    {
+        /// <summary>
+        /// Every rule will have it's own function that will be called here, the rule configuration will determine
+        /// which rules will execute.
+        /// </summary>
+        /// <param name="gitHubEventClient">Authenticated GitHubEventClient</param>
+        /// <param name="prEventPayload">PullRequestEventGitHubPayload deserialized from the json event payload</param>
+        public static async Task ProcessPullRequestEvent(GitHubEventClient gitHubEventClient, PullRequestEventGitHubPayload prEventPayload)
+        {
+            await PullRequestTriage(gitHubEventClient, prEventPayload);
+            ResetPullRequestActivity(gitHubEventClient, prEventPayload);
+            await ResetApprovalsForUntrustedChanges(gitHubEventClient, prEventPayload);
+
+            // After all of the rules have been processed, call to process pending updates
+            int numUpdates = await gitHubEventClient.ProcessPendingUpdates(prEventPayload.Repository.Id, prEventPayload.PullRequest.Number);
```

Comment:
Did you intend to use this numUpdates somewhere?

Reply:
No, we're not using it at the moment outside of testing. I'm just going to ignore the return.